libevmasm/Assembly.cpp

```diff
@@ -1393,6 +1398,24 @@ LinkerObject const& Assembly::assembleEOF() const
 	m_tagPositionsInBytecode = std::vector<size_t>(m_usedTags, std::numeric_limits<size_t>::max());
 	std::map<size_t, uint16_t> dataSectionRef;
 
+	bool storesAuxData = false;
+	bool loadsAuxData = false;
+
+	for (auto& codeSection: m_codeSections)
+	{
+		for (auto const& item: codeSection.items)
+			if (item.type() == AuxDataStore)
+				storesAuxData = true;
+			else if (item.type() == AuxDataLoadN)
+				loadsAuxData = true;
+		if (storesAuxData || loadsAuxData)
+			assertThrow(
+				storesAuxData != loadsAuxData,
+				AssemblyException,
+				"Cannot store and load auxilairy data in the same assembly subroutine."
```

Comment:
```suggestion
				"Cannot store and load auxiliary data in the same assembly subroutine."
```

Reply:
Also, please use `solRequire()` instead of `assertThrow()`.

Reply:
Yes. Sorry. Legacy from `assembleLegacy` :)